Source/Events/EventConverter.cs

```diff
@@ -27,69 +26,231 @@ public EventConverter(IEventTypes eventTypes)
+        public bool TryToSDK(IEnumerable<Contracts.CommittedEvent> source, out CommittedEvents events, out Exception error)
         {
-            var eventType = source.Type.To<EventType>();
-            var clrEventType = _eventTypes.GetTypeFor(eventType);
-            try
+            events = default;
+
+            if (source == null)
             {
-                var content = JsonConvert.DeserializeObject(source.Content, clrEventType);
-                if (source.External)
+                error = new ArgumentNullException(nameof(source));
+                return false;
+            }
+
+            var list = new List<CommittedEvent>();
+            foreach (var sourceEvent in source)
+            {
+                if (!TryToSDK(sourceEvent, out var @event, out error))
                 {
-                    return new CommittedExternalEvent(
-                        source.EventLogSequenceNumber,
-                        source.Occurred.ToDateTimeOffset(),
-                        source.EventSourceId.To<EventSourceId>(),
-                        source.ExecutionContext.ToExecutionContext(),
-                        eventType,
-                        content,
-                        source.Public,
-                        source.ExternalEventLogSequenceNumber,
-                        source.ExternalEventReceived.ToDateTimeOffset());
+                    return false;
                 }
                 else
                 {
-                    return new CommittedEvent(
-                        source.EventLogSequenceNumber,
-                        source.Occurred.ToDateTimeOffset(),
-                        source.EventSourceId.To<EventSourceId>(),
-                        source.ExecutionContext.ToExecutionContext(),
-                        eventType,
-                        content,
-                        source.Public);
+                    list.Add(@event);
                 }
             }
-            catch (Exception ex)
-            {
-                throw new CouldNotDeserializeEvent(
-                    source.Type.Id.To<EventTypeId>(),
-                    clrEventType,
-                    source.Content,
-                    source.EventLogSequenceNumber,
-                    ex);
-            }
+
+            events = new CommittedEvents(list);
+            error = null;
+            return true;
         }
 
         /// <inheritdoc/>
         public CommittedEvents ToSDK(IEnumerable<Contracts.CommittedEvent> source)
-            => new CommittedEvents(source.Select(ToSDK).ToList());
+            => TryToSDK(source, out var events, out var error) ? events : throw error;
+
+        /// <inheritdoc/>
+        public bool TryToSDK(Contracts.CommitEventsResponse source, out CommitEventsResult result, out Exception error)
+        {
+            result = default;
+
+            if (!TryToSDK(source.Events, out var events, out error))
+            {
+                return false;
+            }
```

Comment:
I don't know, but I just like having return something; on the same line of the if-statement if that's the only thing it does
```suggestion
            if (!TryToSDK(source.Events, out var events, out error)) return false;
```